src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandConfiguration.kt

```diff
@@ -129,8 +129,13 @@ open class CargoCommandConfiguration(
 
     @Throws(RuntimeConfigurationException::class)
     override fun checkConfiguration() {
-        if (isRedirectInput && redirectInputPath?.toPath()?.exists() != true) {
-            throw RuntimeConfigurationWarning("Input file doesn't exist")
+        if (isRedirectInput) {
+            val path = redirectInputPath?.toPathOrNull()
+            when {
+                path?.exists() != true -> throw RuntimeConfigurationWarning("Input file doesn't exist")
+                !path.toFile().isFile -> throw RuntimeConfigurationWarning("Input file is not valid")
```

Comment:
Why we need this complicated check?
Do we want to show `Input file is not valid` in case of directory?

Reply:
Yes.